src/main/java/application/rest/v1/configmaps/KindActionMappingProcessor.java

```diff
@@ -0,0 +1,462 @@
+ * This class provides the methods that facilitates a process of getting a configmap for a given
+ * resource following the mapping rules and configmap hierarchy & precedence. 
+ */
+public class KindActionMappingProcessor {
+
+    private static final String className = KindActionMappingProcessor.class.getName();
+
+    // KindActionMapping/KAM definitions
+    private static final String KAM_PLURAL = "kindactionmappings";
+    private static final String KAM_GROUP = "actions.kappnav.io";
+    private static final String KAM_VERSION = "v1";
+    private static final String WILDCARD = "*";
+
+    private static final int MAX_PRECEDENCE = 9;
+    private static final int TOTAL_INDIVIDUAL_MAPPINGS = 4;
+
+    private static final int KSN = 3; // Kind, Subkind, name - instance specific
+    private static final int KN = 2; // Kind, name - instance specific
+    private static final int KS = 1; // Kind, Subkind - subkind specific
+    private static final int K = 0; // Kind - kind specific
+
+    private static final String METADATA_PROPERTY_NAME = "metadata";
+    private static final String NAMESPACE_PROPERTY_NAME = "namespace";
+    private static final String ITEMS_PROPERTY_NAME = "items";
+    private static final String SPEC_PROPERTY_NAME = "spec";
+    private static final String PRECEDENCE_PROPERTY_NAME = "precedence";
+    private static final String MAPPINGS_PROPERTY_NAME = "mappings";
+
+    private static final String APIVERSION_PROPERTY_NAME = "apiVersion";
+    private static final String NAME_PROPERTY_NAME = "name";
+    private static final String SUBKIND_PROPERTY_NAME = "subkind";
+    private static final String KIND_PROPERTY_NAME = "kind";
+    private static final String MAPNAME_PROPERTY_NAME = "mapname";
+
+    private String compNamespace;
+    private String compApiVersion;
+    private String compName;
+    private String compSubkind;
+    private String compKind;
+
+    public KindActionMappingProcessor(String namespace, String apiVersion, String name,
+                                      String subkind, String kind) {
+        this.compNamespace = namespace;
+        this.compApiVersion = apiVersion;
+        this.compName = name;
+        this.compSubkind = subkind;
+        this.compKind = kind;
+    }
+
+    /**
+     * Get configmaps defined in the mappings in the KindActionMapping custom resources
+     * for a given resource. One or more action configmaps may exist to which the same 
+     * resource maps. 
+     * 
+     * @param client
+     * @return configmaps matched the defined action configmap mappings in KAM in order of
+     *         configmap hierarchy & precedence
+     */
+    protected ArrayList<String> getConfigMapsFromKAMs(ApiClient client) {
+
+        if (Logger.isEntryEnabled()) 
+                Logger.log(className, "getConfigMapsFromKAMs", Logger.LogType.ENTRY,"");
+
+        String[][] mapNamesFound = new String[MAX_PRECEDENCE][TOTAL_INDIVIDUAL_MAPPINGS];
```

Comment:
This implementation does not handle multiple KAMs that have the same precedence level.   The spec says: 

```
KindActionMappings resources with the same precedence value are processed together in arbitrary order.
```